doc/usecases.md

```diff
@@ -0,0 +1,188 @@
+#Layer2
+
+Usage can refer to this [doc](layer2.md)
+
+## Prerequisite
+
+The cluster should have at least two nodes, nodeA and nodeB.
+
+## Test endpoint update
+
+- create service test-svc which have one endpoint on nodeA.
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mylbapp
+  labels:
+    app: mylbapp
+spec:
+  containers:
+  - name: helloworld
+    image: karthequian/helloworld
+    imagePullPolicy: IfNotPresent
+  nodeSelector:
+    kubernetes.io/hostname: nodeA
+---
+kind: Service
+apiVersion: v1
+metadata:
+    name:  test-svc
+    annotations:
+        lb.kubesphere.io/v1alpha1: porter
+        #eip.porter.kubesphere.io/v1alpha1: 1.1.1.1 
+        #protocol.porter.kubesphere.io/v1alpha1: layer2 
+spec:
+    selector:
+        app:  mylbapp
+    type:  LoadBalancer 
+    ports:
+      - name:  http
+        port:  80
+        targetPort:  80
+```
+
+- exec 'wget ${eip}' can successfully access to test-svc
+- exec 'ip neigh  show to ${eip} dev ${nic}', we can find lladdr is nodeA's Mac
+- delete the endpoint which on nodeA, and create endpoint on nodeB
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mylbapp
+  labels:
+    app: mylbapp
+spec:
+  containers:
+  - name: helloworld
+    image: karthequian/helloworld
+    imagePullPolicy: IfNotPresent
+  nodeSelector:
+    kubernetes.io/hostname: nodeB
+```
+- exec 'wget ${eip}' can successfully access to test-svc
+- exec 'ip neigh  show to ${eip} dev ${nic}', we can find lladdr is nodeB's Mac
+
+# BGP
+
+Usage can refer to this [doc](simulate_with_bird.md)
+
+## Prerequisite
+
+The cluster should have at least two nodes, nodeA and nodeB.
+
+## Test BGP PortForward
+- config BgpConf with port 17900
+```yaml
+apiVersion: network.kubesphere.io/v1alpha1
+kind: BgpConf
+metadata:
+  name: bgpconf-sample
+spec:
+  # Add fields here
+  as : 65000
+  routerID : 192.168.0.2
+  port: 17900
+```
+- config BgpPeer with usingPortForward
+```yaml
+apiVersion: network.kubesphere.io/v1alpha1
+kind: BgpPeer
+metadata:
+  name: bgppeer-sample
+spec:
+  # Add fields here
+  usingPortForward: true
+  config:
+    peerAs : 65001
+    neighborAddress: 192.168.0.6
+  addAaths:
+    sendMax: 10
+```
+- config bird  neighbour  with port 179
+- exec 'birdctl show protocol', the nighbour's state should be up
+
+## Test BGP passive mode
+- config BgpConf with port 17900
+- config BgpPeer with passiveMode true
+```yaml
+apiVersion: network.kubesphere.io/v1alpha1
+kind: BgpPeer
+metadata:
+  name: bgppeer-sample
+spec:
+  config:
+    peerAs : 65001
+    neighborAddress: 192.168.0.6
+  addAaths:
+    sendMax: 10
+  transport:
+    passiveMode: true
+```
+- config bird neighbour with port 17900
+- exec 'birdctl show protocol', the nighbour's state should be up
+
+## Test endpoint update
+- create service test-svc which have two endpoints on nodeA and nodeB.
```

Comment:
- Create a Service `test-svc` which has two endpoints on nodeA and nodeB respectively.